src/pip/_internal/utils/egg_link.py

```diff
@@ -15,24 +15,31 @@
 ]
 
 
-def _egg_link_name(raw_name: str) -> str:
+def _egg_link_names(raw_name: str) -> List[str]:
     """
     Convert a Name metadata value to a .egg-link name, by applying
     the same substitution as pkg_resources's safe_name function.
     Note: we cannot use canonicalize_name because it has a different logic.
+
+    We also look for the raw name (without normalization) as setuptools 69 changed
+    the way it names .egg-link files (https://github.com/pypa/setuptools/issues/4167).
     """
-    return re.sub("[^A-Za-z0-9.]+", "-", raw_name) + ".egg-link"
+    return [
+        re.sub("[^A-Za-z0-9.]+", "-", raw_name) + ".egg-link",
+        f"{raw_name}.egg-link",
```

Comment:
```suggestion
        re.sub("[^A-Za-z0-9.]+", "-", raw_name) + ".egg-link",
        re.sub("[^A-Za-z0-9._]+", "-", raw_name) + ".egg-link",        
        f"{raw_name}.egg-link",
```

May be this would be better?

Reply:
No. I just made a check and setuptools 69 preserves multiple consecutive `_`too.